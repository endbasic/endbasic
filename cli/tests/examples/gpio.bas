' EndBASIC
' Copyright 2021 Julio Merino
'
' Licensed under the Apache License, Version 2.0 (the "License"); you may not
' use this file except in compliance with the License.  You may obtain a copy
' of the License at:
'
'     http://www.apache.org/licenses/LICENSE-2.0
'
' Unless required by applicable law or agreed to in writing, software
' distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
' WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
' License for the specific language governing permissions and limitations
' under the License.

' Runs the gpio.bas example using mock GPIO data to drive the program.

LOAD "DEMOS:/GPIO.BAS"

DIM __GPIO_MOCK_DATA(20) AS INTEGER
__GPIO_MOCK_DATA(2) = 811 ' Return high for pin 8 (button).
__GPIO_MOCK_DATA(3) = 811 ' Return high for pin 8 (button).
__GPIO_MOCK_DATA(4) = 810 ' Return low for pin 8 (button).
__GPIO_MOCK_LAST = 0

RUN

PRINT "Dumping GPIO trace..."
FOR i = 0 TO __GPIO_MOCK_LAST - 1
    PRINT "__GPIO_MOCK_DATA", i, __GPIO_MOCK_DATA(i)
NEXT
