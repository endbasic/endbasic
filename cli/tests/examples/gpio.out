[2J[1;1H[38;5;11m
 GPIO demo
===========
[39m
This demo showcases how to poll a hardware button attached to a GPIO
pin and how to flash an LED attached to another one.

[38;5;2mTo get started, follow these steps:
[39m
1. Connect an LED to pin 18 and to ground; don't forget to add a
   resistor inline.

2. Connect a push button to pin 8 and to ground.  We'll be
   using the built-in pull-up resistor for the input pin so there is
   no need to do any extra wiring.

[38;5;1mThis demo is only functional on the Raspberry Pi and assumes you have
built EndBASIC with --features=rpi.  If these conditions are not met,
the demo will fail to run.
[39m
[2J[1;1H[38;5;11m
 GPIO demo
===========
[39m
Waiting for a button press on pin 8 ...
Button pressed! Blinking LED on pin 18 ...
5
4
3
2
1
Dumping GPIO trace...
__GPIO_MOCK_DATA        0       803
__GPIO_MOCK_DATA        1       1804
__GPIO_MOCK_DATA        2       811
__GPIO_MOCK_DATA        3       811
__GPIO_MOCK_DATA        4       810
__GPIO_MOCK_DATA        5       1821
__GPIO_MOCK_DATA        6       1820
__GPIO_MOCK_DATA        7       1821
__GPIO_MOCK_DATA        8       1820
__GPIO_MOCK_DATA        9       1821
__GPIO_MOCK_DATA        10      1820
__GPIO_MOCK_DATA        11      1821
__GPIO_MOCK_DATA        12      1820
__GPIO_MOCK_DATA        13      1821
__GPIO_MOCK_DATA        14      1820
